### DISCUSSION TITLE
Disable auto scroll when scrolling to previous data

### DISCUSSION THREAD

**Discussion by an-dew** [patrykandpatrick/vico#951] [Questions] [answered]
### Question

I created a minimal line chart with a dynamic set of data. When new data arrives then chart is correctly scrolled to the latest value (I publish a new point every second).

```
val chartModelProducer = remember { ChartEntryModelProducer() }
chartModelProducer.setEntries(entries)

Chart(
        chart = lineChart(),
        chartModelProducer = chartModelProducer,
        bottomAxis = rememberBottomAxis(),
        endAxis = rememberEndAxis(),
        chartScrollSpec = rememberChartScrollSpec(
            initialScroll = InitialScroll.End,
            autoScrollCondition = AutoScrollCondition.OnModelSizeIncreased,
            isScrollEnabled = true,
        ),
        diffAnimationSpec = snap(),
    )
```
However, when I want to preview previous data and scroll to them, chart scrolls back to the newest point.
Is it possible to temporarily disable auto scrolling when a user checks previous data? 
It will be nice to have a feature which enables auto scroll only when user previews the latest data.

https://github.com/user-attachments/assets/04a1c9ce-dea7-4379-bd18-36c2156e5956

### Vico version(s)

1.14.0

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello, @an-dew.
>Is it possible to temporarily disable auto scrolling when a user checks previous data?

Yes, it is. You can implement it by using a `ScrollListener` that you can register on a `ChartScrollSpec` instance, which is a `Chart` function parameter. Using the `ScrollListener`, you can get the current and maximum scroll values to determine whether the chart is scrolled to the end. You can use this information with `AutoScrollCondition` to disable the auto-scroll conditionally.

Also, calling `chartModelProducer.setEntries(entries)` inside of a `@Composable` function must be avoided. Preferably, you should call it in a `ViewModel` (or its counterpart). If you have to call it inside of a `@Composable` function, please ensure that it is wrapped in e.g. `LaunchedEfect` with an appropriate key to avoid unnecessary data changes on recompositions.
